### PR TITLE
feat(toHaveValue): Enhanced error message with type information on loose equality

### PR DESCRIPTION
**Pull request by mfelmy**
**What**:

As discussed in #218 the error message of `toHaveValue` was enhanced by type information if the received and expected value have loose equality.

**Why**:

To improve DX.

**How**:

The expected and received values are checked for loose equality and enhanced by type information if loose equality is given.

**Checklist**:

- [x] Documentation (N/A)
- [x] Tests
- [ ] Updated Type Definitions (sorry, not so sure where to do that)
- [ ] Ready to be merged (once updating of Type Definitions is done)

If you could point me to where I can update the Type Definitions, I would be happy to do so.

### CODE REVIEW

**Comment by mfelmy** [comment]
Hey guys, could you give me some advice regarding the Type Definitions and why the checks are failing? When I run `npm run validate` locally everything passes and I'm don't really know what to get from the Job log on Travis CI.

**Comment by eps1lon** [comment]
Looks like this only fails with node 13. The branch is also out of date. Try updating it with `master`. 

Then you can investigate by using node 13 locally. Should throw the same error.

**Comment by mfelmy** [comment]
I updated to node 13.11.0 and it still doesn't fail locally. Updated with `master` too.

File: src/__tests__/to-have-value.js [hunk 1]
```diff
@@ -124,6 +124,48 @@ describe('.toHaveValue', () => {
...
+      "<dim>expect(</><red>element</><dim>).toHaveValue(</><green>8</><dim>)</>
+
+      Expected the element to have value:
+      <green>  8 (number)</>
+      Received:
+      <red>  8 (string)</>"
+    `)
+  })
+
+  test('throws type information when the expected input (type=number) value has loose equality with received value', () => {
```
Comment (eps1lon) [inline src/__tests__/to-have-value.js]: I don't think it's currently considering the input type. Just that the expected and actual have loose equality.

Comment (mfelmy) [reply src/__tests__/to-have-value.js]: No it does not indeed. With `input (type=text)` and `input (type=number)` I simply wanted to differentiate the tests.
Maybe renaming it to `'throws with type information when the expected text input value has loose equality with received value'` and `... number input value...`, respectively would be a better option?

Comment (eps1lon) [reply src/__tests__/to-have-value.js]: Yep. Then we don't need separate ones for different input types.

**Comment by mfelmy** [comment]
Sorry to bother again, but when I try to commit the changes the `husky > pre-commit hook` fails.
I don't fully understand why or how to resolve this. To me it seems as if it is due to the formatting of the console output.

Can you give me a hint on how to solve this? Or is it okay to use the `--no-verify` flag in this case?

image

**Comment by eps1lon** [comment]
No idea. I have husky disabled globally anyway.

**Comment by eps1lon** [comment]
@@all-contributors please add @mfelmy for code

**Comment by eps1lon** [comment]
@mfelmy Nice work, thanks!

**Comment by mfelmy** [comment]
Thank you very much for helping me on my first pull request.

**Comment by gnapse** [comment]
@all-contributors please add @mfelmy for code, test

**Comment by gnapse** [comment]
@all-contributors please add @mfelmy for code, test

**Pull request merged**